File: src/main/java/com/designpatterns/dev/behavioral/momento/java/EditorHistory.java
package com.designpatterns.dev.behavioral.momento.java;

import java.util.ArrayList;
import java.util.List;

public class EditorHistory {
    private List<EditorState> stateList = new ArrayList<>();

    public void pushState(EditorState editorState) {
        stateList.add(editorState);
    }

    public EditorState popState() {
        EditorState previousState = stateList.get(getIndex());
        stateList.remove(previousState);
        return stateList.get(getIndex());
    }

    private int getIndex(){
        int size = stateList.size();
        return (size > 0) ? stateList.size() - 1 : 0;
    }
}

File: src/main/java/com/designpatterns/dev/behavioral/momento/kotlin/EditorHistory.kt
package com.designpatterns.dev.behavioral.momento.kotlin

import java.util.*

class EditorHistory {
    private var list: ArrayList<EditorState> = arrayListOf()

    fun pushState(editorState: EditorState){
        list.add(editorState)
    }

    fun popState() : EditorState {
        var editorState = list.get(getIndex())
        list.remove(editorState)
        return list.get(getIndex())
    }

    private fun getIndex() : Int {
        var size = list.size
        return size -1;
    }
}